k8s/upgrade/src/upgrade_resources/upgrade.rs

```diff
@@ -162,13 +164,86 @@ pub struct GetUpgradeArgs {}
 impl GetUpgradeArgs {
     ///  Upgrade the resources.
     pub async fn get_upgrade(&self, namespace: &str) {
-        println!("{namespace}")
-        // To be implemented
+        // Create resources for getting upgrade status
+        match UpgradeEventClient::create_get_upgrade_resource(namespace).await {
+            Ok(_) => {}
+            Err(error) => eprintln! {"Failure {error}"},
+        }
+    }
+}
+
+/// This struct is used to deserialize the output of ugrade events.
+#[derive(Clone, Deserialize)]
+pub(crate) struct UpgradeEvent {
+    from_version: String,
+    to_version: String,
+    message: String,
+}
+
+/// Resource to be created to get upgrade status.
+struct UpgradeEventClient {
+    upgrade_event: Api<Event>,
+}
+
+/// Methods implemented by UpgradeEventClient.
+impl UpgradeEventClient {
+    pub async fn new(ns: &str) -> Result<Self, Error> {
+        let client = Client::try_default().await?;
+        Ok(Self {
+            upgrade_event: Api::<Event>::namespaced(client, ns),
+        })
+    }
+
+    /// Returns an client for upgrade events.
+    pub(crate) fn api_client(&self) -> Api<Event> {
+        self.upgrade_event.clone()
+    }
+
+    /// Fetch upgrade events and print.
+    pub async fn get_and_log(&self, release_name: String) -> Result<(), Error> {
+        let event_lp = ListParams {
+            field_selector: Some(upgrade_event_selector(
+                release_name.as_str(),
+                UPGRADE_JOB_NAME_SUFFIX,
+            )),
+            ..Default::default()
+        };
+
+        let mut event_list = self
+            .api_client()
+            .list(&event_lp)
+            .await?
+            .into_iter()
+            .filter(|e| e.reason == Some(UPGRADE_EVENT_REASON.to_string()))
+            .collect::<Vec<_>>();
+
+        event_list.sort_by(|a, b| b.event_time.partial_cmp(&a.event_time).unwrap());
+        match event_list.get(0) {
+            Some(data) => {
+                let p: UpgradeEvent =
+                    serde_json::from_str(data.message.clone().unwrap().as_str()).unwrap();
```

Comment:
Please handle unwraps? The first one can be done by changing match from Some(data) to Some(Some(data))
and then replacing None => with  _ =>